superset-frontend/src/SqlLab/components/QueryTable/index.tsx

```diff
@@ -29,133 +28,158 @@ import Button from 'src/components/Button';
 import { fDuration } from 'src/modules/dates';
 import Icons from 'src/components/Icons';
 import { Tooltip } from 'src/components/Tooltip';
+import { Query } from 'src/SqlLab/types';
 import ResultSet from '../ResultSet';
 import ModalTrigger from '../../../components/ModalTrigger';
 import HighlightedSql from '../HighlightedSql';
 import { StaticPosition, verticalAlign, StyledTooltip } from './styles';
+import rootReducer from '../../reducers/index';
 
-const propTypes = {
-  columns: PropTypes.array,
-  actions: PropTypes.object,
-  queries: PropTypes.array,
-  onUserClicked: PropTypes.func,
-  onDbClicked: PropTypes.func,
-  displayLimit: PropTypes.number.isRequired,
-};
-const defaultProps = {
-  columns: ['started', 'duration', 'rows'],
-  queries: [],
-  onUserClicked: () => {},
-  onDbClicked: () => {},
-};
+// Acquire redux rootstate type, to be used in useSelector
+type RootState = ReturnType<typeof rootReducer>;
+
+// query's type is original Query; Shallow-copy of query, q's type is QueryTableQuery. So that prop, sql passed to another component will remain string, the type of original Query
+interface QueryTableQueryTemp1 extends Omit<Query, 'sql'> {
+  sql: string | Record<string, any>;
+}
+
+interface QueryTableQueryTemp2 extends Omit<QueryTableQueryTemp1, 'progress'> {
+  progress: number | Record<string, any>;
+}
 
-const openQuery = id => {
+interface QueryTableQuery extends Omit<QueryTableQueryTemp2, 'state'> {
+  state: string | Record<string, any>;
+}
+
+interface QueryTableProps {
+  columns: Array<string>;
+  actions: Record<string, any>;
+  queries: Query[];
+  onUserClicked?: Function;
+  onDbClicked?: Function;
+  displayLimit: number;
+}
+
+const openQuery = (id: number) => {
   const url = `/superset/sqllab?queryId=${id}`;
   window.open(url);
 };
 
-const QueryTable = props => {
+const QueryTable = ({
+  columns = ['started', 'duration', 'rows'],
+  actions,
```

Comment:
I'd suggest destructuring these (see note below)

Reply:
If this is about deconstruction of actions, I have a blocker to do so. I have a same question mentioned above: 

May I ask a question about this comment? Please see comments on line 96 - On line 96 comment, actions are passed in as prop at ResultSet on 259. If deconstruction is made at parameters, there will be no 'actions'. Thus I did deconstruction on line 97, which are currently commented out bc I have errors below. 

And when deconstruction like on line 97 about actions, I got an error => 'Cannot destructure property 'queryEditorSetSql' of 'actions' as it is undefined' , May I have some advice on it? Thank you.

Reply:
Yes, you can keep this here and destructure later if you need the `actions` var to pass them together at once. Since you're also specifying what's needed for each action in the type, then that also helps validate that all the necessary actions are passed in. 
is `'Cannot destructure property 'queryEditorSetSql' of 'actions' as it is undefined'` happening on runtime or typescript compile?

Reply:
The error, 'Cannot destructure property 'queryEditorSetSql' of 'actions' as it is undefined' happens when the test is run. Please see the error message that I have in the attached screenshot for the commented out destructure. 
![errorImg](https://user-images.githubusercontent.com/84494313/150225229-2ba3d6b3-7bf9-4510-a381-ff58809b237f.jpg)
.

Reply:
Super, thank you!